src/components/DateRangeSelect/PDateRangeSelectOptions.vue

```diff
@@ -0,0 +1,87 @@
+<template>
+  <PSelectOptions v-model="value" :options="options" class="p-date-range-select-options">
+    <template #pre-options>
+      <div class="p-1">
+        <PTextInput ref="input" v-model="search" placeholder="Enter a unit like &quot;15&quot;" />
+      </div>
+    </template>
+  </PSelectOptions>
+</template>
+
+<script lang="ts" setup>
+  import { addSeconds, isAfter, isBefore, secondsInDay, secondsInHour, secondsInMinute, secondsInWeek } from 'date-fns'
+  import { computed, onMounted, ref } from 'vue'
+  import PSelectOptions from '@/components/Select/PSelectOptions.vue'
+  import PTextInput from '@/components/TextInput/PTextInput.vue'
+  import { SelectOption } from '@/types'
+  import { toPluralString } from '@/utilities'
+
+  export type DateRangeSelectOptionsValue = number | 'range' | null | undefined
+
+  const props = defineProps<{
+    modelValue: DateRangeSelectOptionsValue,
+    min?: Date,
+    max?: Date,
+  }>()
+
+  const emit = defineEmits<{
+    'update:modelValue': [DateRangeSelectOptionsValue],
+  }>()
+
+  onMounted(() => {
+    input.value?.el?.focus()
+  })
+
+  const value = computed({
+    get() {
+      return props.modelValue
+    },
+    set(value) {
+      emit('update:modelValue', value)
+    },
+  })
+
+  const input = ref<InstanceType<typeof PTextInput>>()
+  const search = ref('')
+
+  const options = computed<SelectOption[]>(() => {
+    const parsed = parseInt(search.value || '1')
+    const unit = isNaN(parsed) ? 1 : parsed
+
+    const spans = [
```

Comment:
I think it makes sense to include week here too.

Reply:
I added weeks and also increased the default threshold for removing an option to 2 months. I just want to avoid things like `100 weeks` showing up.